docs/_snippets/framework/tutorials/external-widget.js

```diff
@@ -0,0 +1,200 @@
+
+		this.isEnabled = isAllowed;
+	}
+}
+
+class ExternalWidget extends Plugin {
+	static get requires() {
+		return [ ExternalWidgetEditing, ExternalWidgetUI ];
+	}
+}
+
+class ExternalWidgetUI extends Plugin {
+	init() {
+		const editor = this.editor;
+
+		editor.ui.componentFactory.add( 'external', locale => {
+			const button = new ButtonView( locale );
+
+			button.set( {
+				label: 'Insert Bitcoin rate',
+				tooltip: true,
+				withText: true,
+				isEnabled: true
+			} );
+
+			button.on( 'execute', () => {
+				editor.execute( 'external' );
+				editor.editing.view.focus();
+			} );
+
+			return button;
+		} );
+	}
+}
+
+class ExternalWidgetEditing extends Plugin {
+	constructor( editor ) {
+		super( editor );
+
+		this.intervalId = this._intervalFetch();
+
+		this.externalDataValue = '';
+	}
+
+	static get requires() {
+		return [ Widget ];
+	}
+
+	destroy() {
+		clearInterval( this.intervalId );
+	}
+
+	init() {
+		this._defineSchema();
+		this._defineConverters();
+		this._updateWidgetData();
+
+		this.editor.commands.add( 'external', new ExternalWidgetCommand( this.editor ) );
+	}
+
+	_intervalFetch() {
+		return setInterval( () => this._updateWidgetData(), 20000 ); // set time interval to 20s
+	}
+
+	async _updateWidgetData( externalUrl = 'https://api2.binance.com/api/v3/ticker/24hr?symbol=BTCUSDT' ) {
+		const response = await fetch( externalUrl );
+		const data = await response.json();
+		const updateTime = new Date( data.closeTime );
+		const parsedData = '$' + Number( data.lastPrice ).toFixed( 2 ) + ' - ' + updateTime.toLocaleString();
+
+		this.externalDataValue = parsedData;
+
+		const rootElement = this.editor.model.document.getRoot();
+
+		for ( const { item } of this.editor.model.createRangeIn( rootElement ) ) {
+			if ( item.is( 'element', 'externalElement' ) ) {
+				this.editor.editing.reconvertItem( item );
+			}
+		}
+	}
+
+	_defineSchema() {
+		const schema = this.editor.model.schema;
+
+		schema.register( 'externalElement', {
+			inheritAllFrom: '$inlineObject',
+			allowAttributes: [ 'data-resource-url' ]
+		} );
+	}
+
+	_defineConverters() {
+		const editor = this.editor;
+
+		editor.conversion.for( 'upcast' ).elementToElement( {
+			view: {
+				name: 'span',
+				attributes: [ 'data-resource-url' ]
+			},
+			model: ( viewElement, { writer } ) => {
+				const externalUrl = viewElement.getAttribute( 'data-resource-url' );
+
+				return writer.createElement( 'externalElement', {
+					'data-resource-url': externalUrl
+				} );
+			}
+		} );
+
+		editor.conversion.for( 'dataDowncast' ).elementToElement( {
+			model: 'externalElement',
+			view: ( modelElement, { writer } ) => {
+				return writer.createEmptyElement( 'span', {
+					'data-resource-url': modelElement.getAttribute( 'data-resource-url' )
+				} );
+			}
+		} );
+
+		editor.conversion.for( 'editingDowncast' ).elementToElement( {
+			model: 'externalElement',
+			view: ( modelElement, { writer: viewWriter } ) => {
```

Comment:
Maybe let's have it consistent with the above.

```suggestion
			view: ( modelElement, { writer } ) => {
```

Reply:
fixed